src/job_shop_scheduler.py

```diff
@@ -0,0 +1,368 @@
+        allow_quadratic_constraints (bool, optional): Whether to allow quadratic constraints. 
+            Defaults to True.
+        out_sol_file (str, optional): Path to the output solution file. Defaults to None.
+        out_plot_file (str, optional): Path to the output plot file. Defaults to None.
+        profile (str, optional): The profile variable to pass to the Sampler. Defaults to None.
+
+    Returns:
+        pd.DataFrame: A DataFrame that has the following columns: Task, Start, Finish, and
+        Resource.
+
+    """    
+    if allow_quadratic_constraints and use_mip_solver:
+        raise ValueError("Cannot use quadratic constraints with MIP solver")
+    model_building_start = time()
+    model = JobShopSchedulingCQM(model_data=job_data, max_makespan=max_makespan)
+    model.define_cqm_model()
+    model.define_variables(job_data)
+    model.add_precedence_constraints(job_data)
+    if allow_quadratic_constraints:
+        model.add_quadratic_overlap_constraint(job_data)
+    else:
+        model.add_disjunctive_constraints(job_data)
+    model.add_makespan_constraint(job_data)
+    model.define_objective_function()
+
+    if verbose:
+        print_cqm_stats(model.cqm)
+    model_building_time = time() - model_building_start
+    solver_start_time = time()
+    if use_mip_solver:
+        sol = model.call_mip_solver(time_limit=solver_time_limit)
+    else:
+        model.call_cqm_solver(time_limit=solver_time_limit, model_data=job_data, profile=profile)
+        sol = model.best_sample
+    solver_time = time() - solver_start_time
+
+    if verbose:
+        print(" \n" + "=" * 55 + "SOLUTION RESULTS" + "=" * 55)
+        print(tabulate([["Completion Time", "Max Possible Make-Span",
+                        "Model Building Time (s)", "Solver Call Time (s)",
+                        "Total Runtime (s)"],
+                        [model.completion_time, 
+                         model.max_makespan,
+                         int(model_building_time), 
+                         int(solver_time),
+                         int(solver_time +  model_building_time)
+                         ]],
+                    headers="firstrow"))
+    
+    # Write solution to a file.
+    if out_sol_file is not None:
+        write_solution_to_file(
+            job_data, model.solution, model.completion_time, out_sol_file)
+
+    # Plot solution
+    if out_plot_file is not None:
+        job_plotter.plot_solution(job_data, model.solution, out_plot_file)
+
+    df = model.solution_as_dataframe()
+    return df
+
+
+if __name__ == "__main__":
+    """Modeling and solving Job Shop Scheduling using CQM solver."""
+
+    # Start the timer
+    start_time = time()
+
+    # Instantiate the parser
+    parser = argparse.ArgumentParser(
+        description='Job Shop Scheduling Using LeapHybridCQMSampler',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument('-instance', type=str,
```

Comment:
I wonder what's the reason for using "X toolkit"-style command options instead of [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html) or [GNU](https://www.gnu.org/prep/standards/html_node/Command_002dLine-Interfaces.html) style (what we use in the rest of the examples)?

Much better fit, IMO, would be something like:
```suggestion
    parser.add_argument('-i', '--instance', type=str,
```